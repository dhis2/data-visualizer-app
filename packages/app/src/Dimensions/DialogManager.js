import React from 'react';
import i18n from '@dhis2/d2-i18n';
import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
} from '@material-ui/core';
import { DataDimension } from './DataDimension';
import { PeriodDimension } from './PeriodDimension';
import { OrgUnitDimension } from './OrgUnitDimension';

// Placeholder for the dimension popup dialogs - using the Options dialog until the components are created
const dimensionComponents = {
    Data: <DataDimension />,
    Period: <PeriodDimension />,
    OrgUnit: <OrgUnitDimension />,
};

export const DialogManager = ({ dialogIsOpen, id, toggleDialog }) => {
    return (
        <Dialog open={dialogIsOpen} onClose={() => toggleDialog(null)}>
            <DialogContent>{dimensionComponents[id]}</DialogContent>
            <DialogActions>
                <Button
                    onClick={() => {
                        console.log('toggled selected state for dimension');
                        toggleDialog(null);
                    }}
                >
                    {i18n.t('Submit')}
                </Button>
                <Button onClick={() => toggleDialog(null)}>
                    {i18n.t('Cancel')}
                </Button>
            </DialogActions>
        </Dialog>
    );
};

export default DialogManager;
