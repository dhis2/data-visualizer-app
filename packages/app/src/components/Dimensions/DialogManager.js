import React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DataDimension from './DataDimension/DataDimension';
import PeriodDimension from './PeriodDimension';
import OrgUnitDimension from './OrgUnitDimension';

export const DialogManager = ({ dialogIsOpen, id, toggleDialog }) => {
    const dimensionComponents = {
        dx: <DataDimension toggleDialog={toggleDialog} />,
        pe: <PeriodDimension toggleDialog={toggleDialog} />,
        ou: <OrgUnitDimension />,
    };
    return id ? (
        <Dialog
            open={dialogIsOpen}
            onClose={() => toggleDialog(null)}
            maxWidth={false}
            disableEnforceFocus
        >
            {dimensionComponents[id]}
        </Dialog>
    ) : null;
};

export default DialogManager;
