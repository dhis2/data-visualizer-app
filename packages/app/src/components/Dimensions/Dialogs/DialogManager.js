import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import debounce from 'lodash-es/debounce';
import isEqual from 'lodash-es/isEqual';

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';

import DataDimension from './DataDimension/DataDimension';
import PeriodDimension from './PeriodDimension/PeriodDimension';
import OrgUnitDimension from './OrgUnitDimension/OrgUnitDimension';
import DynamicDimension from './DynamicDimension/DynamicDimension';
import HideButton from '../../HideButton/HideButton';
import AddToLayoutButton from '../../AddToLayoutButton/AddToLayoutButton';

import { acSetUiActiveModalDialog } from '../../../actions/ui';
import { acSetRecommendedIds } from '../../../actions/recommendedIds';

import {
    sGetUiItemsByDimension,
    sGetUiActiveModalDialog,
} from '../../../reducers/ui';
import { sGetDimensions } from '../../../reducers/dimensions';

import { apiFetchRecommendedIds } from '../../../api/dimensions';

import { FIXED_DIMENSIONS } from '../../../modules/fixedDimensions';

const dxId = FIXED_DIMENSIONS.dx.id;
const peId = FIXED_DIMENSIONS.pe.id;
const ouId = FIXED_DIMENSIONS.ou.id;

export const fixedDialogs = {
    [dxId]: <DataDimension />,
    [ouId]: <OrgUnitDimension />,
    [peId]: <PeriodDimension />,
};

export const defaultState = {
    mounted: [],
};

export class DialogManager extends Component {
    state = defaultState;

    componentDidUpdate = prevProps => {
        const shouldFetchIds =
            !isEqual(prevProps.dxIds, this.props.dxIds) ||
            !isEqual(prevProps.ouIds, this.props.ouIds);

        if (shouldFetchIds) {
            this.fetchRecommended();
        }

        if (
            this.props.dialogId &&
            !this.state.mounted.includes(this.props.dialogId)
        ) {
            this.setState({
                mounted: [...this.state.mounted, this.props.dialogId],
            });
        }
    };

    fetchRecommended = debounce(async () => {
        const ids = await apiFetchRecommendedIds(
            this.props.dxIds,
            this.props.ouIds
        );

        this.props.setRecommendedIds(ids);
    }, 1000);

    renderDialogContent = () => (
        <Fragment>
            {Object.keys(FIXED_DIMENSIONS).map(dimensionId => {
                return this.state.mounted.includes(dimensionId) ? (
                    <div
                        key={dimensionId}
                        style={{
                            display:
                                dimensionId === this.props.dialogId
                                    ? 'block'
                                    : 'none',
                        }}
                    >
                        {fixedDialogs[dimensionId]}
                    </div>
                ) : null;
            })}
            {this.props.dialogId &&
                !Object.keys(FIXED_DIMENSIONS).includes(
                    this.props.dialogId
                ) && (
                    <DynamicDimension
                        dialogId={this.props.dialogId}
                        dialogTitle={
                            this.props.dimensions[this.props.dialogId].name
                        }
                    />
                )}
        </Fragment>
    );

    render() {
        return (
            <Dialog
                open={!!this.props.dialogId}
                onClose={() => this.props.closeDialog(null)}
                maxWidth="lg"
                disableEnforceFocus
                keepMounted
            >
                {this.renderDialogContent()}
                <DialogActions>
                    <HideButton />
                    {this.props.dialogId && (
                        <AddToLayoutButton dialogId={this.props.dialogId} />
                    )}
                </DialogActions>
            </Dialog>
        );
    }
}

DialogManager.propTypes = {
    dialogId: PropTypes.string,
    dxIds: PropTypes.array.isRequired,
    ouIds: PropTypes.array.isRequired,
    closeDialog: PropTypes.func.isRequired,
    setRecommendedIds: PropTypes.func.isRequired,
};

DialogManager.defaultProps = {
    dialogId: null,
};

const mapStateToProps = state => ({
    dialogId: sGetUiActiveModalDialog(state),
    dimensions: sGetDimensions(state),
    dxIds: sGetUiItemsByDimension(state, dxId),
    ouIds: sGetUiItemsByDimension(state, ouId),
});

export default connect(
    mapStateToProps,
    {
        closeDialog: acSetUiActiveModalDialog,
        setRecommendedIds: acSetRecommendedIds,
    }
)(DialogManager);
