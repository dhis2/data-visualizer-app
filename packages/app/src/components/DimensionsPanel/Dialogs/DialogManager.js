import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import debounce from 'lodash-es/debounce';
import isEqual from 'lodash-es/isEqual';

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import i18n from '@dhis2/d2-i18n';
import {
    DataDimension,
    DynamicDimension,
    PeriodDimension,
    OrgUnitDimension,
    ouIdHelper,
    DIMENSION_ID_DATA,
    DIMENSION_ID_PERIOD,
    DIMENSION_ID_ORGUNIT,
    getAxisMaxNumberOfItems,
    getDisplayNameByVisType,
    filterOutFixedDimensions,
} from '@dhis2/analytics';

import HideButton from './HideButton';
import AddToLayoutButton from './AddToLayoutButton/AddToLayoutButton';
import UpdateVisualizationContainer from '../../UpdateButton/UpdateVisualizationContainer';

import {
    acSetUiActiveModalDialog,
    acRemoveUiItems,
    acAddUiItems,
    acSetUiItems,
    acAddParentGraphMap,
} from '../../../actions/ui';
import { acAddMetadata } from '../../../actions/metadata';
import { acSetRecommendedIds } from '../../../actions/recommendedIds';

import {
    sGetUiItems,
    sGetUiItemsByDimension,
    sGetUiActiveModalDialog,
    sGetUiParentGraphMap,
    sGetUiType,
    getAxisIdByDimensionId,
    sGetDimensionIdsFromLayout,
} from '../../../reducers/ui';
import { sGetDimensions } from '../../../reducers/dimensions';
import { sGetMetadata } from '../../../reducers/metadata';
import { sGetSettingsDisplayNameProperty } from '../../../reducers/settings';
import { apiFetchRecommendedIds } from '../../../api/dimensions';
import { removeLastPathSegment, getOuPath } from '../../../modules/orgUnit';
import UpdateButton from '../../UpdateButton/UpdateButton';

export class DialogManager extends Component {
    state = {
        onMounted: false,
    };

    componentDidUpdate = prevProps => {
        const shouldFetchIds =
            !isEqual(prevProps.dxIds, this.props.dxIds) ||
            !isEqual(prevProps.ouIds, this.props.ouIds);

        if (shouldFetchIds) {
            this.fetchRecommended();
        }

        if (
            this.props.dialogId === DIMENSION_ID_ORGUNIT &&
            !this.state.ouMounted
        ) {
            this.setState({ ouMounted: true });
        }
    };

    fetchRecommended = debounce(async () => {
        const ids = await apiFetchRecommendedIds(
            this.props.dxIds,
            this.props.ouIds
        );

        this.props.setRecommendedIds(ids);
    }, 1000);

    onSelect = ({ dimensionId, items }) => {
        this.props.setUiItems({
            dimensionId,
            itemIds: items.map(item => item.id),
        });

        switch (dimensionId) {
            case DIMENSION_ID_ORGUNIT: {
                const forMetadata = {};
                const forParentGraphMap = {};

                items.forEach(ou => {
                    const id = ouIdHelper.removePrefix(ou.id);
                    forMetadata[id] = {
                        id,
                        name: ou.name || ou.displayName,
                        displayName: ou.displayName,
                    };

                    if (ou.path) {
                        const path = removeLastPathSegment(ou.path);

                        forParentGraphMap[ou.id] =
                            path === `/${ou.id}` ? '' : path.replace(/^\//, '');
                    }
                });

                this.props.addMetadata(forMetadata);
                this.props.addParentGraphMap(forParentGraphMap);

                break;
            }
            default: {
                this.props.addMetadata(
                    items.reduce((obj, item) => {
                        obj[item.id] = {
                            id: item.id,
                            name: item.name || item.displayName,
                            displayName: item.displayName,
                        };

                        return obj;
                    }, {})
                );
            }
        }
    };

    getSelectedItems = dialogId => {
        return this.props.selectedItems[dialogId]
            ? this.props.selectedItems[dialogId]
                  .filter(id => this.props.metadata[id])
                  .map((id, index) => ({
                      id,
                      name: this.props.metadata[id].name,
                  }))
            : [];
    };

    getOrgUnitsFromIds = (ids, metadata, parentGraphMap) =>
        ids
            .filter(id => metadata[ouIdHelper.removePrefix(id)] !== undefined)
            .map(id => {
                const ouUid = ouIdHelper.removePrefix(id);
                return {
                    id,
                    name: metadata[ouUid].displayName || metadata[ouUid].name,
                    path: getOuPath(ouUid, metadata, parentGraphMap),
                };
            });

    // The OU content is persisted as mounted in order
    // to cache the org unit tree data
    renderPersistedContent = dimensionProps => {
        const {
            displayNameProperty,
            ouIds,
            metadata,
            parentGraphMap,
            dialogId,
        } = this.props;

        if (this.state.ouMounted) {
            const ouItems = this.getOrgUnitsFromIds(
                ouIds,
                metadata,
                parentGraphMap
            );

            const display =
                DIMENSION_ID_ORGUNIT === dialogId ? 'block' : 'none';

            return (
                <div key={DIMENSION_ID_ORGUNIT} style={{ display }}>
                    <OrgUnitDimension
                        displayNameProperty={displayNameProperty}
                        ouItems={ouItems}
                        {...dimensionProps}
                    />
                </div>
            );
        }

        return null;
    };

    renderDialogContent = () => {
        const {
            displayNameProperty,
            dialogId,
            type,
            dimensions,
            removeUiItems,
            setUiItems,
        } = this.props;

        const dimensionProps = {
            d2: this.context.d2,
            onSelect: this.onSelect,
            onDeselect: removeUiItems,
            onReorder: setUiItems,
        };

        const dynamicContent = () => {
            const selectedItems = this.getSelectedItems(dialogId);
            let infoBoxMessage;

            const axisId = this.props.getAxisIdByDimensionId(dialogId);
            const visType = type;
            const numberOfItems = selectedItems.length;

            const axisMaxNumberOfItems = getAxisMaxNumberOfItems(
                visType,
                axisId
            );

            const hasMaxNumberOfItemsRule = !!axisMaxNumberOfItems;

            if (
                hasMaxNumberOfItemsRule &&
                numberOfItems > axisMaxNumberOfItems
            ) {
                infoBoxMessage =
                    axisMaxNumberOfItems === 1
                        ? i18n.t(
                              `'{{visualiationType}}' is intended to show a single data item. Only the first item will be used and saved.`,
                              {
                                  visualiationType: getDisplayNameByVisType(
                                      visType
                                  ),
                              }
                          )
                        : i18n.t(
                              `'{{visualiationType}}' is intended to show maximum {{maxNumber}} number of items. Only the first {{maxNumber}} items will be used and saved.`,
                              {
                                  visualiationType: getDisplayNameByVisType(
                                      visType
                                  ),
                                  maxNumber: axisMaxNumberOfItems,
                              }
                          );

                selectedItems.forEach((item, index) => {
                    item.isActive = index < axisMaxNumberOfItems;
                });
            }

            if (dialogId === DIMENSION_ID_DATA) {
                return (
                    <DataDimension
                        displayNameProp={displayNameProperty}
                        selectedDimensions={selectedItems}
                        infoBoxMessage={infoBoxMessage}
                        {...dimensionProps}
                    />
                );
            }

            if (dialogId === DIMENSION_ID_PERIOD) {
                return (
                    <PeriodDimension
                        selectedPeriods={selectedItems}
                        {...dimensionProps}
                        // TODO infoBoxMessage should ideally be implemented for all dimensions
                    />
                );
            }

            const dynamicDimensions = filterOutFixedDimensions(
                Object.keys(this.props.dimensions)
            );

            if (dynamicDimensions.includes(dialogId)) {
                const dialogTitle =
                    dimensions[dialogId] && dimensions[dialogId].name;

                return (
                    <DynamicDimension
                        selectedItems={selectedItems}
                        dialogId={dialogId}
                        dialogTitle={dialogTitle}
                        {...dimensionProps}
                        // TODO infoBoxMessage should ideally be implemented for all dimensions
                    />
                );
            }
            return null;
        };

        return (
            <Fragment>
                {this.renderPersistedContent(dimensionProps)}
                {dialogId && dynamicContent()}
            </Fragment>
        );
    };

    renderPrimaryButton = dialogId => (
        <UpdateVisualizationContainer
            renderComponent={handler =>
                this.props.dimensionIdsInLayout.includes(dialogId) ? (
                    <UpdateButton
                        flat
                        size="small"
                        onClick={this.getPrimaryOnClick(handler)}
                    />
                ) : (
                    <AddToLayoutButton
                        onClick={this.getPrimaryOnClick(handler)}
                    />
                )
            }
        />
    );

    getPrimaryOnClick = handler => () => {
        handler();
        this.props.closeDialog(null);
    };

    render() {
        const { dialogId, dimensions } = this.props;
        const keepMounted = !dialogId || dialogId === DIMENSION_ID_ORGUNIT;

        return (
            <Dialog
                data-test="dialog-manager"
                open={dialogId in dimensions}
                onClose={() => this.props.closeDialog(null)}
                maxWidth="lg"
                disableEnforceFocus
                keepMounted={keepMounted}
            >
                {this.renderDialogContent()}
                <DialogActions>
                    <HideButton />
                    {dialogId && this.renderPrimaryButton(dialogId)}
                </DialogActions>
            </Dialog>
        );
    }
}

DialogManager.contextTypes = {
    d2: PropTypes.object,
};

DialogManager.propTypes = {
    dialogId: PropTypes.string,
    dxIds: PropTypes.array,
    ouIds: PropTypes.array.isRequired,
    closeDialog: PropTypes.func.isRequired,
    setRecommendedIds: PropTypes.func.isRequired,
    dimensions: PropTypes.object,
    metadata: PropTypes.object,
    selectedItems: PropTypes.object,
    type: PropTypes.string,
    dimensionIdsInLayout: PropTypes.array.isRequired,
};

DialogManager.defaultProps = {
    dialogId: null,
    dxIds: [],
};

const mapStateToProps = state => ({
    displayNameProperty: sGetSettingsDisplayNameProperty(state),
    dialogId: sGetUiActiveModalDialog(state),
    dimensions: sGetDimensions(state),
    metadata: sGetMetadata(state),
    parentGraphMap: sGetUiParentGraphMap(state),
    dxIds: sGetUiItemsByDimension(state, DIMENSION_ID_DATA),
    ouIds: sGetUiItemsByDimension(state, DIMENSION_ID_ORGUNIT),
    selectedItems: sGetUiItems(state),
    type: sGetUiType(state),
    getAxisIdByDimensionId: dimensionId =>
        getAxisIdByDimensionId(state, dimensionId),
    dimensionIdsInLayout: sGetDimensionIdsFromLayout(state),
});

export default connect(
    mapStateToProps,
    {
        closeDialog: acSetUiActiveModalDialog,
        setRecommendedIds: acSetRecommendedIds,
        setUiItems: acSetUiItems,
        addMetadata: acAddMetadata,
        addUiItems: acAddUiItems,
        removeUiItems: acRemoveUiItems,
        addParentGraphMap: acAddParentGraphMap,
    }
)(DialogManager);
