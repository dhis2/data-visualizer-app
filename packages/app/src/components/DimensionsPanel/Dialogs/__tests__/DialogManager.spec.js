import React from 'react'
import { shallow } from 'enzyme'
import {
    DIMENSION_ID_DATA,
    DIMENSION_ID_PERIOD,
    DIMENSION_ID_ORGUNIT,
} from '@dhis2/analytics'

import { DialogManager } from '../DialogManager'

jest.mock('@material-ui/core/Dialog', () => props => {
    console.log('children', props.children)

    return <div id="mock-mui-dialog">{props.children}</div>
})

jest.mock('@material-ui/core/DialogActions', () => props => (
    <div id="mock-mui-dialog-actions">{props.children}</div>
))

jest.mock('@dhis2/analytics', () => {
    const dataId = 'dx'
    const periodId = 'pe'
    const ouId = 'ou'

    return {
        DataDimension: () => <div />,
        DynamicDimension: () => <div />,
        PeriodDimension: () => <div />,
        OrgUnitDimension: () => <div />,
        isSingleValue: () => true,
        DIMENSION_ID_DATA: dataId,
        DIMENSION_ID_PERIOD: periodId,
        DIMENSION_ID_ORGUNIT: ouId,
        FIXED_DIMENSIONS: {
            [dataId]: {},
            [periodId]: {},
            [ouId]: {},
        },
        getAxisMaxNumberOfItems: () => {},
        filterOutFixedDimensions: () => [],
        getAxisName: () => {},
    }
})

describe('The DialogManager component', () => {
    let props
    let shallowDialog

    const dialogManager = (customProps = {}) => {
        props = {
            ...props,
            ...customProps,
        }

        if (!shallowDialog) {
            shallowDialog = shallow(<DialogManager {...props} />)
        }
        return shallowDialog
    }

    beforeEach(() => {
        props = {
            dialogId: null,
            dimensions: {
                test: {},
            },
            dxIds: ['test'],
            ouIds: [],
            selectedItems: {
                ou: [],
                pe: [],
                dx: ['test'],
            },
            d2: {},
            metadata: {},
            closeDialog: jest.fn(),
            setRecommendedIds: jest.fn(),
            getAxisIdByDimensionId: () => {},
            dimensionIdsInLayout: [],
        }
        shallowDialog = undefined
    })

    it('renders a closed dialog', () => {
        expect(dialogManager()).toMatchSnapshot()
    })

    it('should add the dialogId of fixed dimensions to state "mounted" on first time render', () => {
        const dialog = dialogManager().setProps({
            dialogId: DIMENSION_ID_ORGUNIT,
        })

        expect(dialog.state().ouMounted).toBe(true)
    })

    it('renders the DataDimension content in dialog', () => {
        const dialog = dialogManager().setProps({
            dialogId: DIMENSION_ID_DATA,
        })

        expect(dialog).toMatchSnapshot()
    })

    it('renders the OrgUnitDimension content in dialog', () => {
        const dialog = dialogManager().setProps({
            dialogId: DIMENSION_ID_ORGUNIT,
        })

        expect(dialog).toMatchSnapshot()
    })

    it('renders the PeriodDimension content in dialog', () => {
        const dialog = dialogManager().setProps({
            dialogId: DIMENSION_ID_PERIOD,
        })

        expect(dialog).toMatchSnapshot()
    })

    it('renders OUDimension content with display:none when previously mounted', () => {
        const dialog = dialogManager().setProps({
            dialogId: DIMENSION_ID_ORGUNIT,
        })

        expect(dialog).toMatchSnapshot()

        dialog.setProps({ dialogId: null })
        expect(dialog).toMatchSnapshot()

        dialog.setProps({ dialogId: DIMENSION_ID_DATA })
        expect(dialog).toMatchSnapshot()
    })

    it('sets the recommended Ids (with debounced delay) when a change in dx (Data) or ou (Organisation Unit) occurs', () => {
        const dialog = dialogManager()
        dialog.setProps({ ouIds: ['TEST_OU_ID'] })
        dialog.setProps({ ouIds: ['OTHER_ID'] })

        setTimeout(
            () => expect(props.setRecommendedIds).toHaveBeenCalledTimes(1),
            1001
        )
    })

    it('does not update recommendedIds if other selected ids are udpdated', () => {
        const dialog = dialogManager()
        dialog.setProps({ dimensionIdA: ['itemsByDimensionIdA'] })
        dialog.setProps({
            dimensionIdB: ['itemsByDimensionIdB', 'itemsByDimensionIdC'],
        })

        setTimeout(
            () => expect(props.setRecommendedIds).toHaveBeenCalledTimes(0),
            1001
        )
    })

    it('calls the closeDialog function', () => {
        const dialog = dialogManager().setProps({
            dialogId: DIMENSION_ID_DATA,
        })

        dialog.simulate('close')

        expect(props.closeDialog).toHaveBeenCalled()
    })
})
