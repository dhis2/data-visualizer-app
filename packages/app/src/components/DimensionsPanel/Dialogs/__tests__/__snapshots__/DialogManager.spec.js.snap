// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DialogManager component renders OUDimension content with display:none when previously mounted 1`] = `
<Fragment>
  <Modal
    dataTest="dialog-manager-undefined"
    large={true}
    onClose={[Function]}
    position="top"
  >
    <ModalTitle
      dataTest="dialog-manager-modal-title"
    >
      Organisation units
    </ModalTitle>
    <ModalContent
      dataTest="dialog-manager-modal-content"
    >
      <div
        key="ou"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <OrgUnitDimension
          onSelect={[Function]}
          ouItems={Array []}
        />
      </div>
    </ModalContent>
    <ModalActions
      dataTest="dialog-manager-modal-actions"
    >
      <ButtonStrip
        dataTest="dhis2-uicore-buttonstrip"
      >
        <HideButton
          dataTest="dialog-manager-modal-action-cancel"
          onClick={[Function]}
        />
        <Connect(UpdateVisualizationContainer)
          renderComponent={[Function]}
        />
      </ButtonStrip>
    </ModalActions>
  </Modal>
</Fragment>
`;

exports[`The DialogManager component renders OUDimension content with display:none when previously mounted 2`] = `<Fragment />`;

exports[`The DialogManager component renders OUDimension content with display:none when previously mounted 3`] = `
<Fragment>
  <Modal
    dataTest="dialog-manager-undefined"
    large={true}
    onClose={[Function]}
    position="top"
  >
    <ModalTitle
      dataTest="dialog-manager-modal-title"
    >
      Data
    </ModalTitle>
    <ModalContent
      dataTest="dialog-manager-modal-content"
    >
      <div
        key="ou"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <OrgUnitDimension
          onSelect={[Function]}
          ouItems={Array []}
        />
      </div>
      <DataDimension
        onSelect={[Function]}
        selectedDimensions={Array []}
      />
    </ModalContent>
    <ModalActions
      dataTest="dialog-manager-modal-actions"
    >
      <ButtonStrip
        dataTest="dhis2-uicore-buttonstrip"
      >
        <HideButton
          dataTest="dialog-manager-modal-action-cancel"
          onClick={[Function]}
        />
        <Connect(UpdateVisualizationContainer)
          renderComponent={[Function]}
        />
      </ButtonStrip>
    </ModalActions>
  </Modal>
</Fragment>
`;

exports[`The DialogManager component renders a closed dialog 1`] = `<Fragment />`;

exports[`The DialogManager component renders the DataDimension content in dialog 1`] = `
<Fragment>
  <Modal
    dataTest="dialog-manager-undefined"
    large={true}
    onClose={[Function]}
    position="top"
  >
    <ModalTitle
      dataTest="dialog-manager-modal-title"
    >
      Data
    </ModalTitle>
    <ModalContent
      dataTest="dialog-manager-modal-content"
    >
      <DataDimension
        onSelect={[Function]}
        selectedDimensions={Array []}
      />
    </ModalContent>
    <ModalActions
      dataTest="dialog-manager-modal-actions"
    >
      <ButtonStrip
        dataTest="dhis2-uicore-buttonstrip"
      >
        <HideButton
          dataTest="dialog-manager-modal-action-cancel"
          onClick={[Function]}
        />
        <Connect(UpdateVisualizationContainer)
          renderComponent={[Function]}
        />
      </ButtonStrip>
    </ModalActions>
  </Modal>
</Fragment>
`;

exports[`The DialogManager component renders the OrgUnitDimension content in dialog 1`] = `
<Fragment>
  <Modal
    dataTest="dialog-manager-undefined"
    large={true}
    onClose={[Function]}
    position="top"
  >
    <ModalTitle
      dataTest="dialog-manager-modal-title"
    >
      Organisation units
    </ModalTitle>
    <ModalContent
      dataTest="dialog-manager-modal-content"
    >
      <div
        key="ou"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <OrgUnitDimension
          onSelect={[Function]}
          ouItems={Array []}
        />
      </div>
    </ModalContent>
    <ModalActions
      dataTest="dialog-manager-modal-actions"
    >
      <ButtonStrip
        dataTest="dhis2-uicore-buttonstrip"
      >
        <HideButton
          dataTest="dialog-manager-modal-action-cancel"
          onClick={[Function]}
        />
        <Connect(UpdateVisualizationContainer)
          renderComponent={[Function]}
        />
      </ButtonStrip>
    </ModalActions>
  </Modal>
</Fragment>
`;

exports[`The DialogManager component renders the PeriodDimension content in dialog 1`] = `
<Fragment>
  <Modal
    dataTest="dialog-manager-undefined"
    large={true}
    onClose={[Function]}
    position="top"
  >
    <ModalTitle
      dataTest="dialog-manager-modal-title"
    >
      Period
    </ModalTitle>
    <ModalContent
      dataTest="dialog-manager-modal-content"
    >
      <PeriodDimension
        onSelect={[Function]}
        selectedPeriods={Array []}
      />
    </ModalContent>
    <ModalActions
      dataTest="dialog-manager-modal-actions"
    >
      <ButtonStrip
        dataTest="dhis2-uicore-buttonstrip"
      >
        <HideButton
          dataTest="dialog-manager-modal-action-cancel"
          onClick={[Function]}
        />
        <Connect(UpdateVisualizationContainer)
          renderComponent={[Function]}
        />
      </ButtonStrip>
    </ModalActions>
  </Modal>
</Fragment>
`;
